src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -357,3 +361,327 @@ impl HummockManager {
+        parent_group_id: CompactionGroupId,
+        table_ids: &[StateTableId],
+        partition_vnode_count: u32,
+    ) -> Result<(
+        CompactionGroupId,
+        BTreeMap<CompactionGroupId, Vec<StateTableId>>,
+    )> {
+        if table_ids.is_empty() {
+            return Err(Error::CompactionGroup(
+                "table_ids must not be empty".to_string(),
+            ));
+        }
+
+        if !table_ids.is_sorted() {
+            return Err(Error::CompactionGroup(
+                "table_ids must be sorted".to_string(),
+            ));
+        }
+
+        let parent_table_ids = {
+            let versioning_guard = self.versioning.read().await;
+            versioning_guard
+                .current_version
+                .state_table_info
+                .compaction_group_member_table_ids(parent_group_id)
+                .iter()
+                .map(|table_id| table_id.table_id)
+                .collect_vec()
+        };
+
+        if parent_table_ids == table_ids {
+            return Err(Error::CompactionGroup(format!(
+                "invalid split attempt for group {}: all member tables are moved",
+                parent_group_id
+            )));
+        }
+
+        // move [3,4,5,6]
+        // [1,2,3,4,5,6,7,8,9,10] -> [1,2] [3,4,5,6] [7,8,9,10]
+        // split key
+        // 1. table_id = 3, vnode = 0, epoch = 0
+        // 2. table_id = 7, vnode = 0, epoch = 0
+
+        // The new compaction group id is always generate on the right side
+        // Hence, we return the first compaction group id as the result
+
+        // split 1
+        let mut cg_id_to_table_ids: BTreeMap<u64, Vec<u32>> = BTreeMap::new();
+        let result_vec = self
+            .split_compaction_group_impl(
+                parent_group_id,
+                *table_ids.first().unwrap(),
+                VirtualNode::ZERO,
+            )
+            .await?;
+        assert!(result_vec.len() <= 2);
+
+        let target_compaction_group_id = {
+            if result_vec[0].1.contains(&table_ids[0]) {
+                result_vec[0].0
+            } else {
+                result_vec[1].0
+            }
+        };
+        for (cg_id, table_ids) in result_vec {
+            cg_id_to_table_ids.insert(cg_id, table_ids);
+        }
+
+        // split 2
+        // See the example in L603 and the split rule in `split_compaction_group_impl`.
+        let result_vec = self
+            .split_compaction_group_impl(
+                target_compaction_group_id,
+                *table_ids.last().unwrap(),
+                VirtualNode::MAX,
+            )
+            .await?;
+        assert!(result_vec.len() <= 2);
+        let target_compaction_group_id = {
+            if result_vec[0].1.contains(&table_ids[0]) {
+                result_vec[0].0
+            } else {
+                result_vec[1].0
+            }
+        };
+        for (cg_id, table_ids) in result_vec {
+            cg_id_to_table_ids.insert(cg_id, table_ids);
```

Comment:
Let's add the following assertions:
1. table_ids in different cg are sorted.
2. table_ids in different cg are non-overlapping.
3. the provided table_ids belong to one and only one cg.